### DISCUSSION TITLE
Capture response to file

### DISCUSSION THREAD

**Discussion by tpluscode** [Orange-OpenSource/hurl#1323] [Ideas] [answered]
I have a resource like this: 

```json
{
  "id": "foobar",
  "images": {
    "small": "...url...",
    "medium": "...url...",
    "large": "...url..."
  }
}
```

It would be nice to "capture" entire responses as files within chained requests. I would capture the image URLs and follow the first request with 3 more to get the images

```
GET /resource

HTTP/* 200
[Captures]
id: jsonpath "$.id"
image-small: jsonpath "$.images.small"
image-medium: jsonpath "$.images.medium"
image-large: jsonpath "$.images.large"

GET {{image-small}}

HTTP/* 200
[Captures]
save-as: {{id}}/small.jpg

GET {{image-medium}}

HTTP/* 200
[Captures]
save-as: {{id}}/medium.jpg

GET {{image-large}}

HTTP/* 200
[Captures]
save-as: {{id}}/large.jpg
```

**Comment by jcamiel** [comment]
Hi, 
A possible way of doing this without introducing new syntax is to use [`--output`](https://hurl.dev/docs/manual.html#output) option.
Used on the command line, it allows to dump the last response of a chain to a file. Since a few version, we have `[Options]` section that allows certain specific option to be overridden for a request (see https://hurl.dev/docs/request.html#options).
If we leverage this section and add `output` we could write:

```hurl
GET https://foo/resource

HTTP 200
[Captures]
id: jsonpath "$.id"
image-small: jsonpath "$.images.small"
image-medium: jsonpath "$.images.medium"
image-large: jsonpath "$.images.large"


GET {{image-small}}
[Options]
output: {{id}}/small.jpg
HTTP 200


GET {{image-medium}}
[Options]
output: {{id}}/medium.jpg
HTTP 200


GET {{image-large}}
[Options]
output: {{id}}/large.jpg
HTTP 200
```

(BTW, `HTTP/*` is deprecated in favor of `HTTP`!)

**Reply by jcamiel** [reply]
(I've created an issue for this discussion #1326)

**Reply by RobertGauld** [reply]
I'm having issues with your suggestion:

```
GET ##REDACTED##
[Options]
output: form.html
HTTP 302
```
Results in
```
error: Parsing method
  --> update.hurl:24:1
   |
24 | output: form.html
   | ^ the HTTP method is not valid. Valid values are GET, HEAD, POST, PUT, DELETE, CONNECT, OPTIONS, TRACE, PATCH, LINK, UNLINK, PURGE, LOCK, UNLOCK, PROPFIND or VIEW
   |
```
Commenting out the "output: form.html" line results no errors and the body of the response being printed to STDOUT as expected.

```
$ hurl -V
hurl 3.0.0 libcurl/7.81.0 OpenSSL/3.0.2 zlib/1.2.11 brotli/1.0.3840 zstd/1.4.8 libidn2/2.3.2 libssh/0.9.6/openssl/zlib nghttp2/1.43.0
Features (libcurl):  alt-svc AsynchDNS brotli HSTS HTTP2 IDN IPv6 Largefile libz NTLM NTLM_WB SPNEGO SSL TLS-SRP UnixSockets zstd
Features (built-in): brotli
```

**Reply by jcamiel** [reply]
Hi @RobertGauld , as of Hurl 3.0.0, using `output` in an `[Options]` section is __not yet implemented__, there is an opened issue on it (see https://github.com/Orange-OpenSource/hurl/issues/1326). For the moment, you can just save the output of the last response with stdout.

**Reply by RobertGauld** [reply]
Ah, I misunderstood I thought it was a current workaround not an intention.

**Comment by jcamiel** [comment, marked as answer]
Completed in #1326, `--output` per request will be available in the next Hurl release (should be `4.2.0`)
